pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -127,7 +126,7 @@ func (a Adapter) Push(parameters common.PushParameters) (err error) {
 	var podName string
 
 	// If the component already exists, retrieve the pod's name before it's potentially updated
-	if componentExists {
+	if componentExists && *a.deployment.Spec.Replicas == 1 {
```

Comment:
This is very specific to this issue, and is not generic enough to cover all the cases when a pod is not present.

I made some more reading of this Push() function, and understand now that this call to getPod() is necessary to get the name of the previous pod (the one we synced files to previously), and to compare it later (https://github.com/redhat-developer/odo/pull/5315/files#diff-674dc676ed2f8aa2588ede2db4112fe412bfa0365f7b0f7303d5ae407582d2aaL271) with the pod we will sync the files to.

If the pod is restarted, all the files need to be synced; in the other case, only changed files need to be synced.

Instead of making the specific test on Deployment Replicas, you could react differently if `getPod` in line 131 fails. Instead of returning an error, you could continue and make sure that `podChanged`will be set to true (as it is sure we will need to sync all the files again), for example by keeping `podName` to the empty string.

Also, could you please change the variable name `componentExists` to `deploymentExists`, to make the flow easier to understand?

Reply:
> I made some more reading of this Push() function, and understand now that this call to getPod() is necessary to get the name of the previous pod (the one we synced files to previously), and to compare it later (https://github.com/redhat-developer/odo/pull/5315/files#diff-674dc676ed2f8aa2588ede2db4112fe412bfa0365f7b0f7303d5ae407582d2aaL271) with the pod we will sync the files to.

Yes, that's what I was saying while we were discussing this yesterday. But I was struggling to explain. :) 



> Instead of making the specific test on Deployment Replicas, you could react differently if `getPod` in line 131 fails. Instead of returning an error, you could continue and make sure that `podChanged`will be set to true (as it is sure we will need to sync all the files again), for example by keeping `podName` to the empty string.

But for that to happen, the user will have to wait for whatever the value of `PushTimeout` is in odo preferences. By default, it is 4 minutes. Also, `getPod()` function makes a call to `WaitAndGetPodWithEvents` in `pkg/kclient/pods.go` which mandates the desired phase of the pod to be `Running`: https://github.com/redhat-developer/odo/blob/724f16e689545dd4a81671da3e116a33df4832d3/pkg/kclient/pods.go#L25-L27

Now, I know that we can make change to the code to not mandate it. But I'm honestly having a hard time and would appreciate some help. Using the debugger, I can't seem to be able to get inside below go routine: https://github.com/redhat-developer/odo/blob/724f16e689545dd4a81671da3e116a33df4832d3/pkg/kclient/pods.go#L46-L95